lib/ecto/type.ex

```diff
@@ -291,7 +293,20 @@ defmodule Ecto.Type do
     {:ok, %Ecto.Query.Tagged{value: nil, type: type(type)}}
   end
 
-  def dump({:array, type}, value) do
+  def dump({:embed, %{embed: model}}, %{__struct__: model} = struct) do
```

Comment:
This is going to be moved to the `Ecto.Schema.Serializer.dump`, so it will be easy to share.